### PR TITLE
Add recipe for awherepy

### PR DESCRIPTION
**Pull request by calekochenour**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/awherepy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  skip: True  # [py<38]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - matplotlib
```
Comment (scopatz) [inline recipes/awherepy/meta.yaml:23]: This should be `matplotlib-base`

**Comment by calekochenour** [comment]
Confirming that I am the maintainer of this Python package and willing to be listed in the maintainer section.

File: recipes/awherepy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: b7a1c5a09432f6c1870499bbde8f7ee33396794a9ee1df5196fcc0665bb971a3
+
+build:
+  number: 0
+  skip: True  # [py<38]
```
Comment (scopatz) [inline recipes/awherepy/meta.yaml:14]: Please remove this line and add `noarch: python`.  You can say `python >=3.8` below

File: recipes/awherepy/meta.yaml (hunk 1 above)
Comment (calekochenour) [reply recipes/awherepy/meta.yaml:23]: Copy that.

File: recipes/awherepy/meta.yaml (hunk 2 above)
Comment (calekochenour) [reply recipes/awherepy/meta.yaml:14]: Copy that. Should I specify `python >= 3.8` for both the `host` and `run` sections?

File: recipes/awherepy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+test:
+  imports:
+    - awherepy
+    - awherepy.agronomics
+    - awherepy.crops
+    - awherepy.fields
+    - awherepy.grids
+    - awherepy.plantings
+    - awherepy.models
+    - awherepy.weather
```
Comment (marcelotrevisani) [inline recipes/awherepy/meta.yaml:42]: Could we add ``pip check`` here please?
```suggestion
    - awherepy.weather
  commands:
    - pip check
  requires:
    - pip
```

Comment (calekochenour) [reply recipes/awherepy/meta.yaml:42]: Copy that. Added `pip check`.

**Pull request merged**